command_line/dlsnxs2cbf.py

```diff
@@ -0,0 +1,220 @@
+    result.append("# Threshold_setting: 0 eV")
+    result.append("# Gain_setting: mid gain (vrf = -0.200)")
+    result.append("# N_excluded_pixels = 0")
+    result.append("# Excluded_pixels: badpix_mask.tif")
+    result.append("# Flat_field: (nil)")
+    result.append("# Wavelength %.5f A" % L)
+    result.append("# Detector_distance %.5f m" % (D / 1000.0))
+    result.append("# Beam_xy (%.2f, %.2f) pixels" % (Bx, By))
+    result.append("# Flux 0.000000")
+    result.append("# Filter_transmission %.3f" % A)
+    result.append("# Start_angle %.4f deg." % omega[nn])
+    result.append("# Angle_increment %.4f deg." % omega_increment[nn])
+    result.append("# Detector_2theta 0.0000 deg.")
+    result.append("# Polarization 0.990")
+    result.append("# Alpha 0.0000 deg.")
+    result.append("# Kappa 0.0000 deg.")
+    result.append("# Phi %.4f deg." % phi)
+    result.append("# Phi_increment 0.0000 deg.")
+    result.append("# Omega %.4f deg." % omega[nn])
+    result.append("# Omega_increment %.4f deg." % omega_increment[nn])
+    result.append("# Chi %.4f deg." % chi)
+    result.append("# Chi_increment 0.0000 deg.")
+    result.append("# Oscillation_axis X.CW")
+    result.append("# N_oscillations 1")
+    result.append(";")
+
+    return "\n".join(result)
+
+
+def pack(data):
+    from cbflib_adaptbx import compress
+
+    return compress(data)
+
+
+def make_cbf(in_name, template):
+    f = h5py.File(in_name, "r")
+    depends_on(f)
+
+    global datasets
+
+    import binascii
+
+    start_tag = binascii.unhexlify("0c1a04d5")
+
+    for j in range(len(f["/entry/sample/transformations/omega"][()])):
+        block = 1 + (j // 1000)
+        i = j % 1000
+        header = print_cbf_header(f, j)
+        depth, height, width = f["/entry/data/data_%06d" % block].shape
+        import numpy
+
+        data = flex.int(numpy.int32(f["/entry/data/data_%06d" % block][i]))
+        good = data.as_1d() < 65535
+        data.as_1d().set_selected(~good, -2)
+
+        # set the tile join regions to -1 - MOSFLM cares about this apparently
+        mask = get_mask(width, height)
+        data.as_1d().set_selected(mask.as_1d(), -1)
+
+        compressed = pack(data)
+
+        mime = """
+
+_array_data.data
+;
+--CIF-BINARY-FORMAT-SECTION--
+Content-Type: application/octet-stream;
+     conversions="x-CBF_BYTE_OFFSET"
+Content-Transfer-Encoding: BINARY
+X-Binary-Size: %d
+X-Binary-ID: 1
+X-Binary-Element-Type: "signed 32-bit integer"
+X-Binary-Element-Byte-Order: LITTLE_ENDIAN
+X-Binary-Number-of-Elements: %d
+X-Binary-Size-Fastest-Dimension: %d
+X-Binary-Size-Second-Dimension: %d
+X-Binary-Size-Padding: 4095
+
+""" % (
+            len(compressed),
+            data.size(),
+            data.focus()[1],
+            data.focus()[0],
+        )
+
+        padding = (
+            4095 * chr(0)
+            + """--CIF-BINARY-FORMAT-SECTION----
+;"""
+        )
+
+        with open(template % (j + 1), "wb") as fout:
+            print(template % (j + 1))
```

Comment:
logging call would be better than `print`?

Reply:
Logging not used elsewhere in `dxtbx` - reasonable suggestion but won't be used as a library so not sure worth adding